okio/src/nativeMain/kotlin/okio/DeflaterSink.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio
+
+import platform.zlib.Z_FINISH
+import platform.zlib.Z_NO_FLUSH
+import platform.zlib.Z_SYNC_FLUSH
+
+class DeflaterSink(
+  delegate: Sink,
```

Comment:
Is this the term we use for other transforming sinks and sources?

Reply:
Yep! https://github.com/square/okio/blob/master/okio/src/jvmMain/kotlin/okio/ForwardingSink.kt#L24

Reply:
Yeah but for a forwarding sink it makes sense since it's forwarding all calls to the delegate by default. 

https://github.com/square/okio/blob/e9b7133908e066ca67236ce006460fda9a2269b8/okio/src/jvmMain/kotlin/okio/GzipSink.kt#L42

https://github.com/square/okio/blob/e9b7133908e066ca67236ce006460fda9a2269b8/okio/src/jvmMain/kotlin/okio/DeflaterSink.kt#L46

Seems like just "sink" is used for transforming wrappers.